corehq/apps/api/odata/tests/test_service.py

```diff
@@ -140,3 +140,123 @@ def test_with_case_types(self):
                 ],
             }
         )
+
+
+class TestFormServiceDocument(TestCase, FormOdataTestMixin):
+
+    view_urlname = ODataFormServiceView.urlname
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestFormServiceDocument, cls).setUpClass()
+        cls.client = Client()
+        cls.domain = Domain(name='test_domain')
+        cls.domain.save()
+        cls.web_user = WebUser.create(cls.domain.name, 'test_user', 'my_password')
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain.delete()
+        cls.web_user.delete()
+        super(TestFormServiceDocument, cls).tearDownClass()
+
+    def test_no_credentials(self):
+        response = self.client.get(self.view_url)
+        self.assertEqual(response.status_code, 401)
+
+    def test_wrong_password(self):
+        wrong_credentials = self._get_basic_credentials(self.web_user.username, 'wrong_password')
+        response = self._execute_query(wrong_credentials)
+        self.assertEqual(response.status_code, 401)
+
+    def test_wrong_domain(self):
+        other_domain = Domain(name='other_domain')
+        other_domain.save()
+        self.addCleanup(other_domain.delete)
+        correct_credentials = self._get_correct_credentials()
+        response = self.client.get(
+            reverse(self.view_urlname, kwargs={'domain': other_domain.name, 'app_id': 'my_app_id'}),
+            HTTP_AUTHORIZATION='Basic ' + correct_credentials,
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_missing_feature_flag(self):
+        correct_credentials = self._get_correct_credentials()
+        response = self._execute_query(correct_credentials)
+        self.assertEqual(response.status_code, 404)
+
+    @flag_enabled('ODATA')
+    def test_no_case_types(self):
```

Comment:
i think this name should be `no_forms` or similar

Reply:
updated